site/src/pages/news/20220714-newsletter-4.mdx

```diff
@@ -0,0 +1,169 @@
+
+- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
+  - [1.17.1](https://armeria.dev/release-notes/1.17.1/)
+    - Fixing a bug where Retrofit client calls hang indefinitely
+  - [1.17.0](https://armeria.dev/release-notes/1.17.0/)
+    - Easier way to send and receive RESTful APIs
+    - Ability to limit the maximum length of Thrift messages
+    - Injecting dependencies in annotations
+    - Support for Kotlin nullable type(`?`) in annotated services
+  - [1.16.0](https://armeria.dev/release-notes/1.16.0/)
+      - New metrics for TLS handshake results
+      - Handling <type://Decorator> annotation with <type://GrpcService> for your gRPC stubs
+      - Programmatic <type://Flags> configuration via Java SPI
+  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)
+    - Support for gRPC health check protocol
+    - Multipart file uploads in annotated services
+    - GraphQL support in <type://DocService>
+    - Refactored DNS resolver cache with higher hit ratio and reduced traffic
+    - New metrics that help you get alerts before your TLS certificates are expired
+  - [1.14.1](https://armeria.dev/release-notes/1.14.1/)
+    - Additional logs for <type://EmptyEndpointGroupException>
+  - [1.14.0](https://armeria.dev/release-notes/1.14.0/)
+    - Experimental support for Scala 3
+    - Fluent conversion of <type://HttpResponse> into a desired type using <type://WebClient>
+    - Various build options for gRPC using <type://GrpcClientBuilder>
+  - [1.13.4](https://armeria.dev/release-notes/1.13.4/)
+    - Security fix(CWE-22)
+
+</div>
+<div class="noBullet">
+
+- 👂 Armeria team is hosting [bi-weekly virtual office hours](https://armeria.dev/community/office-hours/).
+  - It was great talking with all of you through the video chat. We hope we can see you again in the near future. :)
+
+</div>
+
+## From the community
+
+<div class="noBullet">
+
+- 🎓 Try out this guided [tutorial](https://armeria.dev/tutorials/rest/blog) for REST Service! This super friendly and simple tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
+  - Service annotations
+  - Request converter
+  - Parameter injection
+  - Exception handler
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions about how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
+  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

Comment:
Oops, I didn't see the entire sentence correctly 😅 Let me revert this change for you.

```suggestion
  - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slides](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

Reply:
I trusted you more than I trusted myself 🤣